python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -519,11 +523,17 @@ def _schedule_runs_at_time(
     tick_context.update_state(TickStatus.SUCCESS)
 
 
-def _get_existing_run_for_request(instance, external_schedule, schedule_time, run_request):
+def _get_existing_run_for_request(
+    instance,
+    external_schedule: ExternalSchedule,
+    schedule_time,
+    run_request,
+):
     tags = merge_dicts(
         PipelineRun.tags_for_schedule(external_schedule),
         {
             SCHEDULED_EXECUTION_TIME_TAG: to_timezone(schedule_time, "UTC").isoformat(),
+            REPOSITORY_LABEL_TAG: external_schedule.get_external_origin().external_repository_origin.get_label(),
```

Comment:
Ah, I more meant that we should put this tag inside of the `PipelineRun.tags_for_schedule` call, but this works if we want to limit the scope and change all the other repository-scoped run fetching calls in one fell swoop.

I imagine we'd need to apply the identical fix for sensors, also.

Reply:
Ignore my comment on `tags_for_schedule`... there's some complication there in terms of extracting the repository label.